### PR TITLE
[Component][Serializer][Normalizer] : Deal it with Has Method for the Normalizer/Denormalizer

### PR DESCRIPTION
**Pull request by jsamouh**
| Q             | A
| ------------- | ---
| Branch?       | 3.4
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | ?
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #23314 
| License       | MIT
| Doc PR        | symfony/symfony-docs

Deal it with Has Method for the Normalizer/Denormalizer

### CODE REVIEW

**Comment by jordscream** [comment]
@Aliance , I stole this mini practice to exercise my Symfony PR ;-) (I'm a newbie)
I will be very interresting if you can comment this PR
Note: For unit test, I did not want to change the existing unit test to add has test method. I prefered to create new unit test with new class

File: src/Symfony/Component/Serializer/Tests/Normalizer/GetSetMethodNormalizerTest.php [hunk 1]
```diff
@@ -497,6 +497,23 @@ public function testPrivateSetter()
         $this->assertEquals('bar', $obj->getFoo());
     }
 
+    public function testHasGetterDenormalize()
+    {
+        $obj = $this->normalizer->denormalize(array('foo' => true), __NAMESPACE__.'\ObjectWithHasGetterDummy');
```
Comment (dunglas) [inline src/Symfony/Component/Serializer/Tests/Normalizer/GetSetMethodNormalizerTest.php]: You can use `ObjectWithHasGetterDummy::class`

File: src/Symfony/Component/Serializer/Tests/Normalizer/GetSetMethodNormalizerTest.php [hunk 2]
```diff
@@ -497,6 +497,23 @@ public function testPrivateSetter()
...
+        $this->assertEquals(true, $obj->hasFoo());
```
Comment (dunglas) [inline src/Symfony/Component/Serializer/Tests/Normalizer/GetSetMethodNormalizerTest.php]: `$this->assertTrue($obj->hasFoo())`

File: src/Symfony/Component/Serializer/Tests/Normalizer/GetSetMethodNormalizerTest.php [hunk 3]
```diff
@@ -807,3 +824,18 @@ public static function setFoo($foo)
         self::$foo = $foo;
     }
 }
+
+class ObjectWithHasGetterDummy
+{
+    private $foo = null;
```
Comment (dunglas) [inline src/Symfony/Component/Serializer/Tests/Normalizer/GetSetMethodNormalizerTest.php]: `= null` is useless, it's the default value.

**Review by dunglas — CHANGES_REQUESTED** [review]
Thanks for this fix! I left some minor comments.

**Comment by jordscream** [comment]
@dunglas , you can check :-)

**Review by Aliance — APPROVED** [review]
Rebase should be done right before merging, not while code review. Otherwise reviewers can not see only made changes.

**Review by dunglas — APPROVED** [review]

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @jordscream.

**Pull request merged**